nomenklatura/enrich/common.py

```diff
@@ -113,9 +114,16 @@ def http_post_json_cached(
                 if rex.response is not None and rex.response.status_code in (401, 403):
                     raise EnrichmentAbort("Authorization failure: %s" % url) from rex
                 if rex.response is not None and rex.response.status_code == 429:
-                    log.info("Rate limit exceeded. Sleeping for 60s.")
-                    time.sleep(61)
-                    return self.http_post_json_cached(url, cache_key, json, cache_days)
+                    if retry > 0:
+                        log.info("Rate limit exceeded. Sleeping for 60s.")
+                        time.sleep(61)
+                        return self.http_post_json_cached(
+                            url, cache_key, json, cache_days, retry - 1
+                        )
+                    else:
+                        raise EnrichmentException(
```

Comment:
There's actually a specific exception `EnrichmentAbort` which is meant to cancel the process upon rate limit. We needed this for OpenCorporates. Can we use it here?

Reply:
sure! I was thinking it might recover later, but perhaps in this case it's better to try again tomorrow.